src/hooks/a11y/useHideAriaSiblings.js

```diff
@@ -0,0 +1,70 @@
+import { useLayoutEffect, useState } from 'react'
+
+/**
+ * Walks up the tree and finds the root parent element just beneath the `body` node.
+ * Returns the root node
+ *
+ * @private
+ *
+ * @param {HTMLElement} elem - the element with the parent to be checked
+ *
+ * @return {HTMLElement}
+ */
+function getRootParent(elem) {
+  const parent = elem.parentElement
```

Comment:
Nit: might be interesting to use iterative approach with same base case. I suppose there's no chance of overflow since there will always be a `document.body`, but a very deeply nested component might have a high space complexity I dunno

Reply:
I might be wrong – but I think in most use cases, the recursive calls should not be an issue. The  tree would need to be fairly deep to cause an overflow, and the use case of this hook would be on a component like the modal that's fairly close to the `body`.

Funny enough – this would only be an issue in ever engine but Webkit. Safari implements TCO in safari 13